src/test/java/com/karankumar/bookproject/backend/service/ReadingGoalServiceTest.java

```diff
@@ -43,4 +56,22 @@ void expectSavingGoalToOverwriteExistingGoal(@NotNull @Autowired ReadingGoalServ
         ReadingGoal actual = goalService.findAll().get(0);
         assertThat(newReadingGoal).isEqualToComparingFieldByField(actual);
     }
+
+    @Test
+    void expectDeleteExistingGoal() {
+        // given
+        assertThat(goalService.count()).isOne();
+
+        // when
+        goalService.delete(existingReadingGoal);
+
+        // then (check that the deletion removed the existingReadingGoal)
+        assertThat(goalService.count()).isZero();
+    }
+
+    private void initReadingGoalServiceTest() {
+        this.existingReadingGoal = new ReadingGoal(20, ReadingGoal.GoalType.BOOKS);
+        goalService.save(this.existingReadingGoal);
+    }
```

Comment:
From a readability perspective, I think it's nicer if this method appears directly underneath the `beforeEach()`. It flows nicer when reading the code instead of jumping around (not too much of an issue now, but this is another maintainability issue)